src/api/scripts/harness_diagnostics.py

```
@@ -0,0 +1,281 @@
+        match = REQUEST_METADATA_PATTERN.search(line)
+        if match:
+            return match.groupdict()
+    return {}
+
+
+def _safe_get_json(url: str, params: dict | None = None) -> tuple[bool, dict | str]:
+    try:
+        response = requests.get(url, params=params, timeout=5)
+        response.raise_for_status()
+        return True, response.json()
+    except Exception as exc:  # pragma: no cover - network best effort
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)